addons/mail/static/src/discuss/web/avatar_card/avatar_card_popover.js

```diff
@@ -30,10 +31,23 @@ export class AvatarCardPopover extends Component {
         this.store = useService("mail.store");
         this.dialog = useService("dialog");
         this.openChat = useOpenChat(this.openChatModel);
+        this.state = useState({ partnerDateTime: "" });
         this.store.fetchStoreData("avatar_card", {
             id: this.props.id,
             model: this.props.model,
         });
+        useDynamicInterval(
+            () => {
+                const localDateTime = this.partner?.localDateTime;
+                if (!localDateTime) {
+                    return;
+                }
```

Comment:
The dependencies must be set as precise as possible for proper updating when they change.
All useful values must be directly read from the callback parameters, not directly from `this`.

In particular, in this case, it might fail to re-set the timer if `localDateTime` is initially empty but then becomes available. It would only re-set it if the partner changes. It's an issue in theory, in practice I can't tell as it also depends what Owl does with the component.

Using callback parameter and making as fine grained as possible (receiving `localDateTime` in this case) ensures proper working in all cases.